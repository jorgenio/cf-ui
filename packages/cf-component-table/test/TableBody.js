import React from 'react';
import renderer from 'react-test-renderer';
import {
  TableBody,
  TableBodyUnstyled,
  TableRowUnstyled
} from '../../cf-component-table/src/index';
import felaTestContext from '../../../felaTestContext';
import { mount } from 'enzyme';

test('should render', () => {
  const component = renderer.create(
    felaTestContext(<TableBody>TableBody</TableBody>)
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('should render extra class names', () => {
  const component = renderer.create(
    felaTestContext(<TableBody className="extra">TableBody</TableBody>)
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('should pass down CSS query props', () => {
  const wrapper = mount(
    <table>
      <TableBodyUnstyled striped hover bordered condensed bare tbodyIndex={0}>
        <TableRowUnstyled />
        <TableRowUnstyled />
      </TableBodyUnstyled>
    </table>
  );

  expect(wrapper.childAt(0).childAt(1).prop('condensed')).toBeTruthy();
  expect(wrapper.childAt(0).childAt(1).prop('striped')).toBeTruthy();
  expect(wrapper.childAt(0).childAt(1).prop('bordered')).toBeTruthy();
  expect(wrapper.childAt(0).childAt(1).prop('hover')).toBeTruthy();
  expect(wrapper.childAt(0).childAt(1).prop('bare')).toBeTruthy();
  expect(wrapper.childAt(0).childAt(1).prop('tbodyIndex')).toBe(0);
  expect(wrapper.childAt(0).childAt(1).prop('rowIndex')).toBe(-1);
});
