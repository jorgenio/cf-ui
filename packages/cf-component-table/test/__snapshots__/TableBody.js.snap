// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<tbody
  className="cf-tf7x23"
>
  TableBody
</tbody>
`;

exports[`should render extra class names 1`] = `
<tbody
  className="extra"
>
  TableBody
</tbody>
`;
